### DISCUSSION TITLE
iBeacon support for ESP32-S3

### DISCUSSION THREAD

**Discussion by dknubben** [arendst/Tasmota#16296] [Q&A] [answered]
[at]DEVs: Please add CONFIG_IDF_TARGET_ESP32S3 to
#if CONFIG_IDF_TARGET_ESP32 || CONFIG_IDF_TARGET_ESP32C3
in tasmota_xsns_sensor/xsns_52_esp32_ibeacon_ble.ino
It's working with this device.

**Comment by barbudor** [comment]
Hello
If you have been able to compile with the change, may be you can do a PR?

**Reply by dknubben** [reply]
Yes, I have compiled it with CONFIG_IDF_TARGET_ESP32S3 and I thinks/I see it's working without problems.
But maybe there are reasons by the devs to not enable it for S3? So I was not sure to make a PR...

**Reply by barbudor** [reply]
Most likely because none from the DEV have a iBeacon and a S3
At least with a PR, some may pickup to validate before it get merged

@sfromis Has you have iBeacon, do you have a S3 and would you be able to try it ?
@Jason2866 I know you have all the boards, do you have an iBeacon ?

I'm planning to order some tonight but that will take some day

**Reply by sfromis** [reply]
Getting ready to try on a S3 board....

**Reply by sfromis** [reply]
I can confirm that with the suggested change on an ESP32-S3, and beacon support activated, I'm seeing both my "real" beacon, and the Android simulated one picked up by Tasmota, as expected. The omission of S3 was probably just because no one asked till now :wink:

**Comment by dknubben** [comment]
For Android there are some iBeacon simulators, so there is no need to have "real" iBeacon hardware for testing

**Reply by sfromis** [reply]
Yeah, just verified that this app can broadcast a beacon picked up by Tasmota:
https://play.google.com/store/apps/details?id=net.alea.beaconsimulator
(remember to first create a beacon, and then back to the front page to activate it)

**Comment by Jason2866** [comment]
I will try this evening or tomorrow.

**Reply by Jason2866** [reply]
@dknubben or @sfromis Since i never used what do i have to enable in Tasmota to fetch a ibeacon?

**Reply by sfromis** [reply]
Assuming a build matching `tasmota32-bluetooth.bin`:
```
SetOption115 1
iBeacon 1
BLEAddrFilter 3
```
Less noisy operation if you also assign a BLEAlias to the MAC for the beacon, and `iBeaconOnlyAliased 1`, without that you may see a lot of "crap" looking "interesting", but useless.

**Reply by Jason2866** [reply]
Confirmed working.

**Comment by dknubben** [comment]
@Jason2866
Good question!
That was also for me not very clear.

console:
BLEAddrFilter 3
only for debugging:
Weblog 3
perhaps:
reboot 1
iBeacon 1
BLEdetails3 [shown MAC]
ex. output:
BLE = {"DetailsBLE":{"mac":"5A37F355364E/1","RSSI":-86,"p":"0303AAFE1516AAFE00BFA43DC59E93893F6D3D1D000000000000","0xfeaa":"00BFA43DC59E93893F6D3D1D000000000000"}}

**Reply by sfromis** [reply]
Note: Reading DetailsBLE is **not** the place to look for a successfully detected beacon. You should instead expect something like this for my Android simulated beacon:
`{"Time":"2022-08-22T22:29:57.742+02:00","IBEACON":{"UID":"5BF71B47ED964903912AED51540FE1D0","MAJOR":"0000","MINOR":"0000","MAC":"730B5170F7B0","RSSI":-29,"STATE":"ON","PERSEC":1}}`

**Reply by dknubben** [reply]
Yes, sorry, you are right

**Comment by Jason2866** [comment, marked as answer]
@dknubben  PR #16311
